coral-schema/src/main/java/com/linkedin/coral/schema/avro/ViewToAvroSchemaConverter.java

```diff
@@ -92,6 +92,57 @@ public Schema toAvroSchema(String dbName, String tableOrViewName) {
     return avroSchema;
   }
 
+  /**
+   * This is the main API to generate the avro schema String for a given Dali view
+   *
+   * @param dbName
+   * @param tableOrViewName
+   * @param strictMode if strictMode is set to True, we will not fall back to Hive schema if avro.schema.literal
+   *                        is missing in table properties. In addition, original namespace in base tables will be preserved.
+   *                   if strictMode is set to False, we will fall back to Hive schema if avro.schema.literal
+   *                        is missing in table properties. A new set of namespace will be generated for
+   *                        the resulting schema. The rule is as follows:
+   *                                1. Top level namespace is dbName.tableOrViewName
+   *                                2. Nested namespace is parentNamespace.parentFieldName
+   * @param isPrettyPrint if pretty print string is required
+   *
+   * @return avro schema String for a given Dali view [dbName, viewName]
+   */
+  public String toAvroSchemaString(String dbName, String tableOrViewName, boolean strictMode, boolean isPrettyPrint) {
+    return toAvroSchema(dbName, tableOrViewName, strictMode).toString(isPrettyPrint);
+  }
+
+  /**
+   * This is the main API to generate the avro schema String for a given Dali view
+   *
+   * @param dbName
+   * @param tableOrViewName
+   * @param strictMode if strictMode is set to True, we will not fall back to Hive schema if avro.schema.literal
+   *                        is missing in table properties. In addition, original namespace in base tables will be preserved.
+   *                   if strictMode is set to False, we will fall back to Hive schema if avro.schema.literal
+   *                        is missing in table properties. A new set of namespace will be generated for
+   *                        the resulting schema. The rule is as follows:
+   *                                1. Top level namespace is dbName.tableOrViewName
+   *                                2. Nested namespace is parentNamespace.parentFieldName
+   *
+   * @return avro schema String for a given Dali view [dbName, viewName]
+   */
+  public String toAvroSchemaString(String dbName, String tableOrViewName, boolean strictMode) {
+    return toAvroSchemaString(dbName, tableOrViewName, strictMode, false);
+  }
+
+  /**
+   * This is the main API to generate the avro schema String for a given Dali view on strict mode
+   *
+   * @param dbName
+   * @param tableOrViewName
+   *
+   * @return avro schema String for a given Dali view [dbName, viewName]
+   */
+  public String toAvroSchemaString(String dbName, String tableOrViewName) {
+    return toAvroSchemaString(dbName, tableOrViewName, false, false);
+  }
+
```

Comment:
Can we remove this? I do no think it is needed. Explicit guidance on strict mode is required I think.

Reply:
will remove